### PR TITLE
integrate freertos into app

### PR DESCRIPTION
**Pull request by adambujak**
I was looking at the architecture of the application and it's gotten too complex. hopefully we can reduce some of this complexity by using freertos.

though i am worried we will run low on memory

### CODE REVIEW

**Comment by stephen99scott** [comment]
@adambujak Is this a draft PR now or an actual PR?

**Comment by adambujak** [comment]
@stephen99scott it's an actual one

**Comment by stephen99scott** [comment]
I'm getting some warnings compiling this
image

**Comment by stephen99scott** [comment]
Looks like an annoying issue with using hal and ll, I guess we can just ignore it for the time being

**Comment by adambujak** [comment]
@stephen99scott 

yeah I tried a few things to get rid of it.
couldn't figure anything out

File: firmware/applications/SignBuddy/task_config.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#ifndef TASK_CONFIG_H
+#define TASK_CONFIG_H
+
+// higher number is higher priority
+#define sensors_TASK_PRIORITY    (tskIDLE_PRIORITY + 4)
```
Comment (stephen99scott) [inline firmware/applications/SignBuddy/task_config.h:5]: caps

**Review by stephen99scott — APPROVED** [review]

File: firmware/applications/SignBuddy/task_config.h (hunk 1 above)
Comment (adambujak) [reply firmware/applications/SignBuddy/task_config.h:5]: it's changed in the next pr.

File: firmware/applications/SignBuddy/Makefile [hunk 2]
```diff
@@ -29,8 +42,10 @@ HAL_SOURCES = \
 	$(HAL_ROOT)/Src/stm32l0xx_ll_usart.c \
 	$(HAL_ROOT)/Src/stm32l0xx_hal.c \
 	$(HAL_ROOT)/Src/stm32l0xx_hal_cortex.c \
+	$(HAL_ROOT)/Src/stm32l0xx_hal_rcc.c \
```
[the remark this answers is not in the document]
Comment (adambujak) [reply firmware/applications/SignBuddy/Makefile:43]: I can't remember what it was needed for but it was

[the remark this answers is not in the document]
Comment (adambujak) [reply firmware/applications/SignBuddy/Makefile:43]: not the tsc though

[the remark this answers is not in the document]
Comment (stephen99scott) [reply firmware/applications/SignBuddy/Makefile:43]: Okay, no problem

**Pull request merged**